### DISCUSSION TITLE
how to fix PD constants

### DISCUSSION THREAD

**Discussion by Rukendu** [google-deepmind/mujoco#391] [Asking for Help]
I am trying to simulate a quadruped. My plan is to swing 2 legs at a time and the other 2 will be on stance/fix by that time. Now at the onset of the swinging phase, there is a lag of control and the body moves back a little. I am using PD controller. I suspect that it is due to the kp,kv values. The lag was larger when these values were smaller and it kept getting smaller as I increased these values. How to fix this thing?

**Comment by kevinzakka** [comment]
I realize this may not directly answer your question, but one thing to note is that in MuJoCo, it is more stable to apply damping to the joints as opposed to adding a velocity actuator. See [here](https://mujoco.readthedocs.io/en/latest/modeling.html#damping) for more information.

A video of the behavior would be helpful to understand the context better and a model would be even better :)

**Comment by yuvaltassa** [comment]
There is no way to completely "fix" this, if I understand what you are asking.
Physics is second-order: acceleration -> velocity -> position, so there will always be a lag between the applied force and position changes, in MuJoCo and in the real world.
